src/seccomp/mod.rs

```diff
@@ -139,6 +140,33 @@ impl FilterContext {
 
         Ok(())
     }
+
+    pub fn notify_fd(&self) -> Result<Option<i32>> {
+        let res = unsafe { seccomp_notify_fd(self.ctx) };
+        // -1 indicates the notify fd is not set. This can happen if no seccomp
+        // notify filter is set.
+        if res == -1 {
+            return Ok(None);
+        }
+
+        if res == -(libc::EINVAL as i32) {
+            bail!("invalid seccomp context used to call notify fd");
+        }
+
+        if res == -(libc::EFAULT as i32) {
+            bail!("internal libseccomp fault; likely no seccomp filter is loaded");
+        }
+
+        if res == -(libc::EOPNOTSUPP as i32) {
+            bail!("seccomp notify filter not supported");
+        }
+
+        if res > 0 {
+            return Ok(Some(res));
+        }
```

Comment:
How about using `match res`?

Reply:
I thought about it. However, note the return code is actually negative, so it is comparison to an expression instead of a value. Match expression doesn't really support this unless I am using match guard. At that point, Using if statement here is likely easier to read.

Reply:
I undetstand. How about using `nix::errorno`?
https://docs.rs/nix/0.22.1/nix/errno/enum.Errno.html#method.from_i32

Reply:
That's a good idea.